### DISCUSSION TITLE
What if we renamed "createFrom" to "createOnBehalfOf"?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#51] [Ideas]
The former is a bit unclear with respect to what exactly is the role of `from`: is it the stream sender, or the stream funder?

`createOnBehalfOf` would be more verbose, but much clearer.

**Comment by razgraf** [comment]
I love the `onBehalfOf` terminology.
A choice I've seen Uniswap make is to only have base functions (create/mint etc.) with a `recipient` argument and handle the logic inside - not separate methods e.g. `mint/mintOnBehalf`.

**Reply by PaulRBerg** [reply]
That's an interesting approach! It wouldn't work for `create` and `createWithDuration`, but it might work for `createFor`.

**Comment by PaulRBerg** [comment]
Locking, there is no `createFrom` function anymore.
